test/scriptingService.test.ts

```diff
@@ -73,7 +72,88 @@ suite('Scripting Service Tests', () => {
         scriptingService = new ScriptingService(connectionManager.object);
-        const script = await scriptingService.scriptSelect(testNode, 'test_uri');
+        let scriptingParams = scriptingService.createScriptingParams(testNode, 'test_uri', ScriptOperation.Select);
+        const scriptingObject = scriptingService.getObjectFromNode(testNode);
+        assert.notEqual(scriptingParams, undefined);
+        assert.equal(scriptingParams.scriptDestination, 'ToEditor');
+        assert.equal(scriptingParams.scriptingObjects[0].name, scriptingObject.name);
+        assert.equal(scriptingParams.scriptingObjects[0].schema, scriptingObject.schema);
+        assert.equal(scriptingParams.scriptingObjects[0].type, scriptingObject.type);
+        assert.equal(scriptingParams.operation, ScriptOperation.Select);
+    });
+
+    test('Test Script Select function', async () => {
+        const testNodeMetadata: ObjectMetadata = {
+            metadataType: MetadataType.Table,
+            metadataTypeName: 'Table',
+            urn: undefined,
+            schema: 'dbo',
+            name: 'test_table'
+        };
+        const testNode = new TreeNodeInfo('test_table (System Versioned)', undefined, undefined,
+            undefined, undefined, 'Table', undefined, undefined, undefined, testNodeMetadata);
+        scriptingService = new ScriptingService(connectionManager.object);
+        const script = await scriptingService.script(testNode, 'test_uri', ScriptOperation.Select);
+        assert.notEqual(script, undefined);
+    });
+
+    test('Test Script Create function', async () => {
+        const testNodeMetadata: ObjectMetadata = {
+            metadataType: MetadataType.Table,
+            metadataTypeName: 'Table',
+            urn: undefined,
+            schema: 'dbo',
+            name: 'test_table'
+        };
+        const testNode = new TreeNodeInfo('test_table (System Versioned)', undefined, undefined,
+            undefined, undefined, 'Table', undefined, undefined, undefined, testNodeMetadata);
+        scriptingService = new ScriptingService(connectionManager.object);
+        const script = await scriptingService.script(testNode, 'test_uri', ScriptOperation.Create);
+        assert.notEqual(script, undefined);
+    });
+
+    test('Test Script Execute function', async () => {
+        const testNodeMetadata: ObjectMetadata = {
+            metadataType: MetadataType.SProc,
+            metadataTypeName: 'StoredProcedure',
+            urn: undefined,
+            schema: 'dbo',
+            name: 'test_proc'
+        };
+        const testNode = new TreeNodeInfo('test_table (System Versioned)', undefined, undefined,
+            undefined, undefined, 'Table', undefined, undefined, undefined, testNodeMetadata);
+        scriptingService = new ScriptingService(connectionManager.object);
+        const script = await scriptingService.script(testNode, 'test_uri', ScriptOperation.Execute);
```

Comment:
It would be more useful to ensure that the correct path is being taken by also checking that the mock SqlToolsServiceClient is getting the correct params sent for the calls to sendRequest. Right now this is only vaguely checking that things are returning as expected.